word_language_model/main.py

```diff
@@ -145,11 +145,11 @@ def evaluate(data_source):
             data, targets = get_batch(data_source, i)
             if args.model == 'Transformer':
                 output = model(data)
+                output = output.view(-1, ntokens)
             else:
                 output, hidden = model(data, hidden)
                 hidden = repackage_hidden(hidden)
-            output_flat = output.view(-1, ntokens)
-            total_loss += len(data) * criterion(output_flat, targets).item()
+            total_loss += len(data) * criterion(output, targets).item()
```

Comment:
Should it be `criterion(output_flat, targets)`?

Reply:
I modified the `RNNModel` so that it returns the `LogSoftMax` of flatten output. 
Since the transformer requires flattening, it was handled in the if condition. 

Although this can be done in other ways. It was a quick fix since it was producing negative losses for RNN models.
We can import `log_softmax` in `main.py` and add another if statement. If it's RNN, just adding `criterion(log_softmax(output_flat, dim=-1), targets)` would also do the job. 
`log_softmax` was already imported in `model.py`, so I didn't want to import in `main.py` too.